zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -449,13 +449,23 @@ where
     ) -> Self {
         // Prevent loss of miner funds due to an unsupported or incorrect address type.
         if let Some(miner_address) = mining_config.miner_address.clone() {
-            assert_eq!(
-                miner_address.network(),
-                network.clone(),
-                "incorrect miner address config: {miner_address} \
-                         network.network {network} and miner address network {} must match",
-                miner_address.network(),
-            );
+            match network.kind() {
+                network_kind @ NetworkKind::Mainnet => assert_eq!(
```

Comment:
If we refactor the error message below, we can simplify:

```suggestion
                NetworkKind::Mainnet => assert_eq!(
```

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/8383/commits/fb2d13a7062d5d65db49cf89a472425c9efd59bd.